### ISSUE TITLE
protocol detection error when running behind nginx

### ISSUE THREAD

**Issue by gtech99** [Intermesh/groupoffice#235]
In file: go/modules/core/core/model/Settings.php

method detectURL, doesn't detect https properly when GO is running on port 80 behind nginx which is serving SSL pages

**Comment by mschering** [comment]
I think GO will be unable to detect this situation. There could even be multiple proxies.
I think it needs to be solved manually.

Just out of curiosity, why this setup?

**Comment by gtech99** [comment]
Well, this is a fairly standard setup with nginx for many reasons. We don't like to expose the application servers to the Internet.

In any case, how about adding some extra fields check to the code ?

Something like this

$isSecure = false;
if (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') {
    $isSecure = true;
}
elseif (!empty($_SERVER['HTTP_X_FORWARDED_PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https' || !empty($_SERVER['HTTP_X_FORWARDED_SSL']) && $_SERVER['HTTP_X_FORWARDED_SSL'] == 'on') {
    $isSecure = true;
}
$REQUEST_PROTOCOL = $isSecure ? 'https' : 'http';

**Comment by mschering** [comment]
Are you able to test that code? If you could do that then I'm happy to merge it into the code.

**Comment by gtech99** [comment]
Yes, I can confirm it works.

In GO code, simply change the https line to this:

$https = (isset($_SERVER["HTTPS"]) && ($_SERVER["HTTPS"] == "on" || $_SERVER["HTTPS"] == "1")) || !empty($_SERVER["HTTP_X_SSL_REQUEST"] || (!empty($_SERVER['HTTP_X_FORWARDED_
PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https') || (!empty($_SERVER['HTTP_X_FORWARDED_SSL']) && $_SERVER['HTTP_X_FORWARDED_SSL'] == 'on'));

on line 53

**Comment by gtech99** [comment]
thanks! I think it's just a formatting error in GIT. Now I noticed even SERVER['HTTP_X_FORWARDED' is in italics. Let me try again

$https = (isset($_SERVER["HTTPS"]) && ($_SERVER["HTTPS"] == "on" || $_SERVER["HTTPS"] == "1")) || !empty($_SERVER["HTTP_X_SSL_REQUEST"] || (!empty($_SERVER['HTTP_X_FORWARDED_
PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] == 'https') || (!empty($_SERVER['HTTP_X_FORWARDED_SSL']) && $_SERVER['HTTP_X_FORWARDED_SSL'] == 'on'));
